src/main/java/com/digitalasset/testing/junit4/Sandbox.java

```diff
@@ -156,36 +161,15 @@ public Sandbox build() {
           logLevel);
     }
 
-    private SandboxBuilder() {}
-  }
-
-  private final boolean useReset;
-
-  private Sandbox(
-      Path projectRoot,
-      Optional<String> testModule,
-      Optional<String> testStartScript,
-      Duration waitTimeout,
-      String[] parties,
-      Path darPath,
-      BiConsumer<DamlLedgerClient, ManagedChannel> setupApplication,
-      boolean useWallclockTime,
-      boolean useReset,
-      Optional<String> ledgerId,
-      Optional<LogLevel> logLevel) {
-    this.sandboxManager =
-        new SandboxManager(
-            projectRoot,
-            testModule,
-            testStartScript,
-            waitTimeout,
-            parties,
-            darPath,
-            setupApplication,
-            useWallclockTime,
-            ledgerId,
-            logLevel);
-    this.useReset = useReset;
+    private void validate() {
+      require(
+          darPath != null && Files.isRegularFile(darPath),
+          String.format("DAR path '%s' must point to a file.", darPath));
```

Comment:
nit again :)
i disagree with this assumption. what we need is that we can _open and read_ this file. this condition is not satisfying nor required: what if perms missing or if it's not a regular file but a pipe etc. so I vote on these options:
- remove the require
- check if the file is readable instead